tests/Unit/NumericalAlgorithms/Spectral/Test_Projection.cpp

```diff
@@ -0,0 +1,258 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cstddef>
+#include <numeric>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "NumericalAlgorithms/Spectral/LegendreGaussLobatto.hpp"
+#include "NumericalAlgorithms/Spectral/Projection.hpp"
+
+namespace {
+DataVector apply_matrix(const Matrix& m, const DataVector& v) noexcept {
+  ASSERT(m.columns() == v.size(), "Bad apply_matrix");
+  DataVector result(m.rows(), 0.);
+  for (size_t i = 0; i < m.rows(); ++i) {
+    for (size_t j = 0; j < m.columns(); ++j) {
+      result[i] += m(i, j) * v[j];
+    }
+  }
+  return result;
+}
+
+double inner_product(const DataVector& a, const DataVector& b) noexcept {
+  ASSERT(a.size() == b.size(), "Sizes differ");
+  const DataVector temp = a * b * Basis::lgl::quadrature_weights(a.size());
```

Comment:
I think this could be replaced by a call to `definite_integral(a * b, Index<1>{a.size()})`.